books/models.py

```diff
@@ -10,16 +11,31 @@ def get_default_owner():
     )
 
 
+class LoanStatus(ChoiceEnum):
+    AV = "available"
+    OL = "on loan"
+    RQ = "requested"
+    NA = "not available"
+
+
 class Book(models.Model):
     title = models.CharField(max_length=100)
     author = models.CharField(max_length=100)
     cover = models.ImageField(upload_to="covers/", blank=True)
     thumb = models.ImageField(upload_to="covers/", blank=True)
     published_date = models.DateTimeField(blank=True, null=True)
     last_updated = models.DateTimeField(auto_now_add=True, null=True)
+    status = models.CharField(
+        max_length=50,
+        choices=[(tag.name, tag.value) for tag in LoanStatus],
+        default=(LoanStatus.AV.name),
+    )
     owner = models.ForeignKey(
         get_user_model(), on_delete=models.SET(get_default_owner)
     )
+    holder = models.ManyToManyField(
+        get_user_model(), related_name="holder", through="BookHolder"
```

Comment:
it's better to use a plural name here, as this is used for reverse lookups, so `books_held` could be more appropriate!

Reply:
Also since this is many to many it refers to many holders, the field should be called "holders".

Reply:
I've changed these

Reply:
oops just saw this was outdated?